Resources/doc/first_steps.rst

```diff
@@ -249,6 +249,226 @@ to easily fetch objects based on multiple conditions:
         ['price' => 'ASC']
     );
 
+Automatically Fetching Objects (DocumentValueResolver)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 4.6
+
+    The support of this feature was added in Doctrine MongoDB ODM Bundle 4.6.
+
+In many cases, you can use the ``DocumentValueResolver`` to do the query for
+you automatically! You can simplify the controller to:
+
+.. code-block:: php
+
+    // src/Controller/ProductController.php
+    namespace App\Controller;
+
+    use App\Document\Product;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Routing\Annotation\Route;
+    // ...
+
+    #[Route('/product/{id}')]
+    public function showAction(Product $product): Response
+    {
+        // use the Product!
+        // do something, like pass the $product object into a template
+    }
+
+That's it! The bundle uses the ``{id}`` from the route to query for the ``Product``
+by the ``id`` column. If it's not found, a 404 page is generated.
+
+This behavior is enabled by default on all your controllers. You can
+disable it by setting the ``doctrine_mongodb.controller_resolver.auto_mapping``
+config option to ``false``.
+
+When disabled, you can enable it individually on the desired controllers by
+using the ``MapDocument`` attribute:
+
+.. code-block:: php
+
+    // src/Controller/ProductController.php
+    namespace App\Controller;
+
+    use App\Document\Product;
+    use Doctrine\Bundle\MongoDBBundle\Attribute\MapDocument;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Routing\Annotation\Route;
+    // ...
+
+    class ProductController extends AbstractController
+    {
+        #[Route('/product/{id}')]
+        public function show(
+            #[MapDocument]
+            Product $product
+        ): Response {
+            // use the Product!
+            // ...
+        }
+    }
+
+.. tip::
+
+    When enabled globally, it's possible to disable the behavior on a specific
+    controller, by using the ``MapDocument`` set to ``disabled``:
+
+.. code-block:: php
+
+        public function show(
+            #[CurrentUser]
+            #[MapDocument(disabled: true)]
+            User $user
+        ): Response {
+            // User is not resolved by the DocumentValueResolver
+            // ...
+        }
+
+Fetch Automatically
+~~~~~~~~~~~~~~~~~~~
+
+If your route wildcards match properties on your document, then the resolver
+will automatically fetch them:
+
+.. code-block:: php
+
+    /**
+     * Fetch via primary key because {id} is in the route.
+     */
+    #[Route('/product/{id}')]
+    public function showByPk(Post $post): Response
```

Comment:
MongoDB does not have a `primary key` concept, should we just say `identifier` or is there more to itfunctionality-wise?

Reply:
The method name still uses `Pk` part :)

Reply:
Oops, thanks! fixed